modules/@apostrophecms/schema/index.js

```diff
@@ -595,6 +588,45 @@ module.exports = {
       name: 'group' // visual grouping only
     });
 
+    self.addFieldType({
+      name: 'range',
+      vueComponent: 'AposInputRange',
+      convert: async function (req, field, data, object) {
+        object[field.name] = self.apos.launder.float(data[field.name], field.def, field.min, field.max);
+        if (field.required && (_.isUndefined(data[field.name]) || !data[field.name].toString().length)) {
+          throw self.apos.error('required');
+        }
+        if (data[field.name] && isNaN(parseFloat(data[field.name]))) {
+          throw self.apos.error('invalid');
+        }
+        // Allow for ranges to go unset
+        // `min` here does not imply requirement, it is the minimum value the range UI will represent
+        if (!data[field.name] || data[field.name] < parseFloat(field.min)) {
+          object[field.name] = null;
+        }
+      },
+      validate: function (field, options, warn, fail) {
+        if (!field.min) {
+          fail('Property "min" must be set.');
+        }
+        if (!field.max) {
+          fail('Property "max" must be set.');
+        }
+        if (isNaN(parseFloat(field.max))) {
+          fail('Property "max" must be an integer.');
+        }
+        if (isNaN(parseFloat(field.min))) {
+          fail('Property "min" must be an integer.');
+        }
+        if (field.step && isNaN(parseFloat(field.step))) {
+          fail('Property "step" must be an integer.');
```

Comment:
You're testing for float, but the message is integer. We should allow float values as well.

Reply:
Also, why are we parsing here? It should be an actual float. We don't need to accept strings for numeric values as part of schema field definitions.

Reply:
So typeof should be `number` rather than doing any parsing.

Reply:
I was being soft on this because technically range inputs accept strings buuuutt we do not need to support both, clamped these down